File: server/fix_tests.py
#!/usr/bin/env python3
import re

# Read the file
with open('cast/handlers_test.go', 'r') as f:
    content = f.read()

# Replace the pattern
pattern = r'(\s+)store := NewJobStore\(\)\s+handler := NewHandler\(store\)'
replacement = r'\1store := NewJobStore()\n\1deviceStore := NewDeviceStore()\n\1handler := NewHandler(store, deviceStore)'

content = re.sub(pattern, replacement, content)

# Write back
with open('cast/handlers_test.go', 'w') as f:
    f.write(content)

print("Fixed all NewHandler calls")

File: server/fix_tests.go
package main

import (
	"fmt"
	"os"
	"regexp"
)

func main() {
	// Read the file
	content, err := os.ReadFile("cast/handlers_test.go")
	if err != nil {
		fmt.Printf("Error reading file: %v\n", err)
		os.Exit(1)
	}

	// Replace the pattern
	re := regexp.MustCompile(`(\s+)store := NewJobStore\(\)\s+handler := NewHandler\(store\)`)
	result := re.ReplaceAllString(string(content), `${1}store := NewJobStore()
${1}deviceStore := NewDeviceStore()
${1}handler := NewHandler(store, deviceStore)`)

	// Write back
	err = os.WriteFile("cast/handlers_test.go", []byte(result), 0644)
	if err != nil {
		fmt.Printf("Error writing file: %v\n", err)
		os.Exit(1)
	}

	fmt.Println("Fixed all NewHandler calls")
}
